src/game/resource.c

```diff
@@ -42,42 +95,93 @@ workshop_type resource_to_workshop_type(resource_type resource)
     }
 }
 
-int inventory_is_set(int inventory, int flag)
+void resource_init(void)
+{
+    int food_index = 0;
+    int good_index = 0;
+
+    for (int i = RESOURCE_MIN; i < RESOURCE_MAX_LEGACY; i++) {
+        resource_data *info = &resource_info[resource_mappings[0][i]];
+        info->text = lang_get_string(23, i);
+        info->image.cart.single_load = image_group(GROUP_FIGURE_CARTPUSHER_CART) + 8 * i;
+        if (resource_is_food(info->type)) {
+            info->image.cart.multiple_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_FOOD) + 8 * food_index;
+            info->image.cart.eight_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_FOOD) + 32 + 8 * (food_index + 1);
+            food_index++;
+        } else {
+            info->image.cart.multiple_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_RESOURCE) + 8 * good_index;
+            info->image.cart.eight_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_RESOURCE) + 8 * good_index;
+            good_index++;
+        }
+        info->image.empire = image_group(GROUP_EMPIRE_RESOURCES) + i;
+        info->image.icon = image_group(GROUP_RESOURCE_ICONS) + i;
+        info->image.storage = image_group(GROUP_BUILDING_WAREHOUSE_STORAGE_FILLED) + 4 * (i - 1);
+        info->image.editor.icon = image_group(GROUP_EDITOR_RESOURCE_ICONS) + i;
+        info->image.editor.empire = image_group(GROUP_EDITOR_EMPIRE_RESOURCES) + i;
+    }
+
+    fish_resource.text = lang_get_string(CUSTOM_TRANSLATION, TR_RESOURCE_FISH);
+    fish_resource.image.cart.single_load = image_group(GROUP_FIGURE_CARTPUSHER_CART) + 696;
+    fish_resource.image.cart.multiple_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_FOOD) + 56;
+    fish_resource.image.cart.eight_loads = image_group(GROUP_FIGURE_CARTPUSHER_CART_MULTIPLE_FOOD) + 88;
+    fish_resource.image.storage = image_group(GROUP_BUILDING_WAREHOUSE_STORAGE_FILLED) + 60;
+    fish_resource.image.icon = image_group(GROUP_RESOURCE_ICONS) + 17;
+    fish_resource.image.empire = image_group(GROUP_EMPIRE_RESOURCES) + 17;
+    fish_resource.image.editor.icon = image_group(GROUP_EDITOR_RESOURCE_ICONS) + 17;
+    fish_resource.image.editor.empire = image_group(GROUP_EDITOR_EMPIRE_RESOURCES) + 17;
+
+    resource_info[RESOURCE_DENARII].image.icon = image_group(GROUP_RESOURCE_ICONS) + 16;
+    resource_info[RESOURCE_DENARII].text = lang_get_string(23, 16);
+
+    resource_info[RESOURCE_TROOPS].text = lang_get_string(23, 17);
+}
+
+const resource_data *resource_get_data(resource_type resource)
 {
-    return (inventory >> flag) & 1;
+    if (resource == RESOURCE_MEAT && scenario_building_allowed(BUILDING_WHARF)) {
+        return &fish_resource;
+    }
+    return &resource_info[resource];
 }
 
-void inventory_set(int *inventory, int flag)
+void resource_set_mapping(int version)
```

Comment:
On the contrary: when a game with an old resource mapping is loaded, it will automatically be upgraded to the latest resource mapping, precisely by using this function.

The idea of `resource_set_mapping` is to find the correspondence between the savegame's original mapping and the mapping currently used by the game, so the resource ID's can be properly updated.

Reply:
The current mapping is still mapping zero, that's why there is nothing else.